### DISCUSSION TITLE
Multiply but do not touch the destination transparency

### DISCUSSION THREAD

**Discussion by NetSysFire** [ImageMagick/ImageMagick#5290] [Help] [answered]
I have a little strand of grass that is meant to be colored:

![image](https://user-images.githubusercontent.com/59517351/177022319-21e69b4f-00ab-46e7-8e9b-c288d5bb7fab.png)

So I am multiplying the image with green but the entire image gets tinged green.

```bash
composite -size 16x16 -compose multiply 'xc:#2D5A27' grass.png grass_colored.png
```

I only want to touch the non-transparent areas. I already played around with different modes such as `dst_in` but these did not produce the results I wanted. Yes, there is a decent chance that I just used the tool wrong because I am very much a noob in all of this. I could swear the command above worked in a different way some months ago, so any help on this is appreciated.

**Comment by fmw42** [comment, marked as answer]
You do not say what is your Imagemagick version or platform/OS. So try this. It is Unix syntax for Imagemagick 6.

```
convert 177022319-21e69b4f-00ab-46e7-8e9b-c288d5bb7fab.png +write mpr:img +delete \
\( mpr:img -alpha extract \) \
\( mpr:img -alpha off -size 16x16 xc:"#2D5A27" -compose multiply -composite \) \
+swap -alpha off -compose copy_opacity -composite grass_colored.png
```

![grass_colored](https://user-images.githubusercontent.com/13458218/177024363-4e7b15af-5a94-4694-b7b5-18ee3fc63289.png)

For Imagemagick 7 Unix syntax, change convert to magick.

**Reply by NetSysFire** [reply]
>You do not say what is your Imagemagick version or platform/OS.

Oh, sorry. My bad. This is 7.1.0.37-1 on Linux. Thank you very much for the quick answer!

**Comment by urban-warrior** [comment]
For IMv7, utilize masks:

```
magick 177022319-21e69b4f-00ab-46e7-8e9b-c288d5bb7fab.png -channel-fx "alpha=>read-mask" -channel rgb -size 16x16 -compose multiply 'xc:#2D5A27' -composite grass_colored.png
```

**Reply by fmw42** [reply]
I believe that he means "magick" not "convert", unless you have a symbolic link.

**Reply by NetSysFire** [reply]
Interestingly enough the distribution I am on, Arch Linux, has symbolic links for the utilities:

```bash
$ pacman -Ql imagemagick | grep "/usr/bin"
imagemagick /usr/bin/
imagemagick /usr/bin/Magick++-config
imagemagick /usr/bin/MagickCore-config
imagemagick /usr/bin/MagickWand-config
imagemagick /usr/bin/animate
imagemagick /usr/bin/compare
imagemagick /usr/bin/composite
imagemagick /usr/bin/conjure
imagemagick /usr/bin/convert
imagemagick /usr/bin/display
imagemagick /usr/bin/identify
imagemagick /usr/bin/import
imagemagick /usr/bin/magick
imagemagick /usr/bin/magick-script
imagemagick /usr/bin/mogrify
imagemagick /usr/bin/montage
imagemagick /usr/bin/stream
```

All of them except the first three (and `magick` of course) are symbolic links to `magick`.

**Reply by NetSysFire** [reply]
Interestingly enough the distribution I am on, Arch Linux, has symbolic links for the utilities:

```bash
$ pacman -Ql imagemagick | grep "/usr/bin"
imagemagick /usr/bin/
imagemagick /usr/bin/Magick++-config
imagemagick /usr/bin/MagickCore-config
imagemagick /usr/bin/MagickWand-config
imagemagick /usr/bin/animate
imagemagick /usr/bin/compare
imagemagick /usr/bin/composite
imagemagick /usr/bin/conjure
imagemagick /usr/bin/convert
imagemagick /usr/bin/display
imagemagick /usr/bin/identify
imagemagick /usr/bin/import
imagemagick /usr/bin/magick
imagemagick /usr/bin/magick-script
imagemagick /usr/bin/mogrify
imagemagick /usr/bin/montage
imagemagick /usr/bin/stream
```

All of them except the first three (and `magick` of course) are symbolic links to `magick`.

**Reply by urban-warrior** [reply]
Correct.  We encourage version 7 users to utilize the `magick` utility, however, we include the legacy utilities to support legacy command-lines as users transition from version 6 to 7.  Since the legacy utilities are simple symbolic links, they have an insignificant footprint.
